### PR TITLE
Support vector type in AArch64 C abi

### PR DESCRIPTION
**Pull request by yuyichao**
And add more tests about aarch64 specific features (`__fp16`, `__int128`, homogeneous short vector aggregate).

### CODE REVIEW

File: src/abi_aarch64.cpp [hunk 1]
```diff
@@ -16,11 +16,53 @@ namespace {
...
+        break;
+    case 16:
+        lltype = T_vec128;
+        break;
+    default:
+        return nullptr;
+    }
+    // Since `dt` is pointer free and has no padding and is 8 or 16 in size.
+    // `ft0` must be concrete, immutable with no padding and we don't need
+    // to check if it's size is legal since it is automatically done by
```
Comment (tkelman) [inline src/abi_aarch64.cpp]: its size

File: src/abi_aarch64.cpp [hunk 2]
```diff
@@ -16,11 +16,53 @@ namespace {
...
+    case 8:
+        lltype = T_vec64;
+        break;
+    case 16:
+        lltype = T_vec128;
+        break;
+    default:
+        return nullptr;
+    }
+    // Since `dt` is pointer free and has no padding and is 8 or 16 in size.
```
Comment (eschnett) [inline src/abi_aarch64.cpp]: `size.` -> `size,`

File: src/ccalltest.c [hunk 3]
```diff
@@ -393,16 +397,82 @@ JL_DLLEXPORT void *test_echo_p(void *p) {
...
 
-JL_DLLEXPORT __m128 test_m128(__m128 a, __m128 b, __m128 c, __m128 d ) {
+JL_DLLEXPORT __m128 test_m128(__m128 a, __m128 b, __m128 c, __m128 d )
+{
     return _mm_add_ps(a, _mm_mul_ps(b, _mm_sub_ps(c, d)));
 }
 
 #endif
+
+#ifdef _CPU_AARCH64_
```
Comment (Keno) [inline src/ccalltest.c]: Why are these tests aarch64 specific?

Comment (yuyichao) [inline src/ccalltest.c]: `__int128`, `__fp16` and the vector types used below are AArch64 extensions (especially when passed as arguments or return values). I'm pretty sure at least `__fp16` is not allowed as return value elsewhere (even when similar types is defined in C).

**Pull request merged**